PhysicsTools/PatAlgos/plugins/PATGenJetSlimmer.cc

```diff
@@ -74,8 +78,17 @@ pat::PATGenJetSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iSet
     auto mapping = std::make_unique<std::vector<int> >();
     mapping->reserve(src->size());
 
+    unsigned nl = 0; // number of loose jets
     for (View<reco::GenJet>::const_iterator it = src->begin(), ed = src->end(); it != ed; ++it) {
-        if (!cut_(*it)) {
+
+	bool selectedLoose = false;
+	if ( nLoose_ > 0 && cutLoose_(*it) ) {
```

Comment:
No reason to compute it when nl>nLoose:

unsigned nl = nLoose;  // number of remaining loose jets to consider
...
bool selectedLoose = false;
if (nl>0) { selectedLoose = true; --nl; }
bool pass = cut_(*it) || selectedLoose